fog/ledger/enclave/api/src/lib.rs

```diff
@@ -38,22 +42,31 @@ pub struct OutputContext {
     pub merkle_root_block: u64,
 }
 
-/// An intermediate struct for holding data required to check key images for the
-/// client. This is returned by `client_check_key_images` and allows untrusted
-/// to gather data that will be encrypted for the client in
-/// `outputs_for_client`.
-///
-/// Eventually we will do the key image check in ORAM, but for now untrusted
-/// will do the check directly.
+/// We will do the key image check in ORAM, but for now untrusted
 #[derive(Clone, Debug, Default, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
-pub struct KeyImageContext {
-    pub key_images: Vec<KeyImage>,
+pub struct UntrustedKeyImageQueryResponse {
+    /// The number of blocks at the time that the request was evaluated.
+    pub highest_processed_block_count: u64,
+
+    /// The timestamp of the highest processed block at the time that the
+    /// request was evaluated.
+    pub highest_processed_block_signature_timestamp: u64,
+
+    /// The index of the last known block, which can be obtained by calculating
+    /// last_known_block_count - 1. We don't store the index but instead store a
+    /// count so that we have a way of representing no known block (0).
+    pub last_known_block_count: u64,
+
+    /// The cumulative txo count of the last known block.
+    pub last_known_block_cumulative_count: u64,
 }
 
 /// The API for interacting with a ledger node's enclave.
 pub trait LedgerEnclave: ReportableEnclave {
     // UTILITY METHODS
 
+    fn new(&self, logger: Logger) -> Self;
```

Comment:
I don't think we should add this, is it needed?

Reply:
Different implementations may need to have different arguments to construct them, so there won't be a common API for this. It seems like you marked all implementations `todo!` anyways

Reply:
done